File: DSA/SAMAGRA_590014891/Day_1_Question_1.c
#include <stdio.h>

void replaceNegativesWithZero(int nums[], int size) {
    for (int i = 0; i < size; i++) {
        if (nums[i] < 0) {
            nums[i] = 0;
        }
    }
}

void printArray(int nums[], int size) {
    for (int i = 0; i < size; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
}

int main() {
    int nums1[] = {-1, 2, -3, 4, 5};
    int size1 = sizeof(nums1) / sizeof(nums1[0]);
    replaceNegativesWithZero(nums1, size1);
    printArray(nums1, size1); 

    int nums2[] = {0, -5, -10};
    int size2 = sizeof(nums2) / sizeof(nums2[0]);
    replaceNegativesWithZero(nums2, size2);
    printArray(nums2, size2);  

    int nums3[] = {1, 2, 3, 4};
    int size3 = sizeof(nums3) / sizeof(nums3[0]);
    replaceNegativesWithZero(nums3, size3);
    printArray(nums3, size3);  

    return 0;
}

File: DSA/N_PARTH_PANDEY_590012761/Day1Question1.py
def replace_negatives_with_zero(nums):
    for i in range(len(nums)):
        if nums[i] < 0:
            nums[i] = 0

def print_array(nums):
    for num in nums:
        print(num, end=' ')
    print()

def main():
    nums1 = [-1, 2, -3, 4, 5]
    replace_negatives_with_zero(nums1)
    print_array(nums1)

    nums2 = [0, -5, -10]
    replace_negatives_with_zero(nums2)
    print_array(nums2)

    nums3 = [1, 2, 3, 4]
    replace_negatives_with_zero(nums3)
    print_array(nums3)

if __name__ == "__main__":
    main()
